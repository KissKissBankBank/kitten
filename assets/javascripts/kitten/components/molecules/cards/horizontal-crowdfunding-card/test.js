import React from 'react'
import renderer from 'react-test-renderer'
import { HorizontalCrowdfundingCard } from './index'

describe('<HorizontalCrowdfundingCard />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <HorizontalCrowdfundingCard
            title="Custom title"
            description="Custom description"
            href="#"
            imageProps={{ src: '#image', alt: '' }}
            noProgressOnMobile
            progress={50}
            progressColor="transparent"
            info={(<span>Test info</span>)}
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
