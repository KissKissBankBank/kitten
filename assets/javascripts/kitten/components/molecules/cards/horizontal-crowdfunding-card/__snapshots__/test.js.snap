// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalCrowdfundingCard /> by default matches with snapshot 1`] = `
<a
  className="sc-bwzfXH buWGgf k-HorizontalCrowdfundingCard k-HorizontalCrowdfundingCard--noProgressOnMobile"
  href="#"
>
  <div
    className="k-HorizontalCrowdfundingCard__imageWrapper"
  >
    <img
      alt=""
      src="#image"
    />
  </div>
  <h3
    className="k-u-size-tiny k-u-weight-regular k-HorizontalCrowdfundingCard__title"
  >
    Custom title
  </h3>
  <p
    className="k-u-size-micro k-u-weight-light k-HorizontalCrowdfundingCard__description"
  >
    Custom description
  </p>
  <div
    className="k-HorizontalCrowdfundingCard__info"
  >
    <span>
      Test info
    </span>
  </div>
  <div
    className="k-HorizontalCrowdfundingCard__progress"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={0}
      className="sc-bdVaJa iqopMg k-Meters-Progress k-HorizontalCrowdfundingCard__progress__ramp k-Meters-Progress--andromeda"
      role="progressbar"
    >
      <div
        className="k-Meters-Progress__ramp"
        style={
          Object {
            "height": 4,
          }
        }
      />
    </div>
    <span
      className="k-u-size-micro k-u-weight-light k-HorizontalCrowdfundingCard__progress__value"
    >
      50
      Â %
    </span>
  </div>
</a>
`;
